frontend/asset_mapper.rst

```diff
@@ -0,0 +1,1030 @@
+file. Use the ``debug:asset-map`` command to see all valid logical paths
+in your app.
+
+More likely, you're importing the failing asset from a CSS file (e.g.
+``@import url('other.css')``) or a JavaScript file:
+
+.. code-block:: javascript
+
+    // assets/controllers/farm-controller.js
+    import '../farm/chicken.js';
+
+When doing this, the path should be *relative* to the file that's importing it
+(and, in JavaScript files, should start with ``./`` or ``../``). In this case,
+``../farm/chicken.js`` would point to ``assets/farm/chicken.js``. To
+see a list of *all* invalid imports in your app, run:
+
+.. code-block:: terminal
+
+    $ php bin/console cache:clear
+    $ php bin/console debug:asset-map
+
+Any invalid imports will show up as warnings on top of the screen (make sure
+you have ``symfony/monolog-bundle`` installed):
+
+.. code-block:: text
+
+    WARNING   [asset_mapper] Unable to find asset "../images/ducks.png" referenced in "assets/styles/app.css".
+    WARNING   [asset_mapper] Unable to find asset "./ducks.js" imported from "assets/app.js".
+
+Missing Asset Warnings on Commented-out Code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+AssetMapper looks in your JavaScript files for ``import`` lines so that it can
+:ref:`automatically add them to your importmap <automatic-import-mapping>`.
+This is done via regex and works very well, though it isn't perfect. If you
+comment-out an import, it will still be found and added to your importmap. That
+doesn't harm anything, but could be surprising.
+
+If the imported path cannot be found, you'll see warning log when that asset
+is being built, which you can ignore.
+
+.. _deployment:
+
+Deploying AssetMapper
+---------------------
+
+When you're ready to deploy, just be sure to "compile" your assets during
+deployment:
+
+.. code-block:: terminal
+
+    $ php bin/console asset-map:compile
+
+That's it! This will write all your assets into the ``public/assets/`` directory,
+along with a few JSON files so that the ``importmap`` can be rendered lightning fast.
+
+But to make sure your site is performant, be sure that your web server
+(or a proxy) is running HTTP/2, is compressing your assets and setting
+long-lived Expires headers on them. See :ref:`Optimization <optimization>` for
+more details.
+
+.. _optimization:
+
+Optimizing Performance
+----------------------
+
+To make your AssetMapper-powered site fly, there are a few things you need to
+do. If you want to take a shortcut, you can use a service like `Cloudflare`_,
+which will automatically do most of these things for you:
+
+- **Use HTTP/2**: Your web server **must** be running HTTP/2 so the browser
+  can download assets in parallel. HTTP/2 is automatically enabled in Caddy
+  and can be activated in Nginx and Apache. Or, proxy your site through a
+  service like Cloudflare, which will automatically enable HTTP/2 for you.
+
+- **Compress your assets**: Your web server should compress your assets
```

Comment:
```suggestion
- **Compress your assets**: Your web server should compress (e.g. using gzip) your assets
```

Given "compress" and "minify" is often mixed up in a front-end pipeline, it might be a good idea to make sure we're not meaning Caddy automatically minifies your asset files :)